sympy/utilities/tests/test_pytest.py

```diff
@@ -61,3 +61,21 @@ def test_unexpected_exception_is_passed_through_with():
 
 def test_second_argument_should_be_callable_or_string():
     raises(TypeError, lambda: raises("irrelevant", 42))
+
+
+def test_assert_raise_message():
+    import re
+
+    message1 = re.compile(r"division by zero", re.IGNORECASE)
+    message2 = re.compile(r"zero", re.IGNORECASE)
+    message3 = re.compile(r"one", re.IGNORECASE)
+    message4 = re.compile(r"some error message", re.IGNORECASE)
+
+    assert_raise_message(ZeroDivisionError, message1, lambda: 1/0) # error message: 'division by zero'
+    assert_raise_message(ZeroDivisionError, message2, lambda: 1/0)
+
+    raises(AssertionError, lambda: assert_raise_message(ZeroDivisionError, message3, lambda: 1/0))
+
+    #with assert_raise_message(ValueError, message4) as exc_wrapper:
+    #    raise ValueError("some error message")
+    #assert isinstance(exc_wrapper, ValueError)
```

Comment:
@ylemkimon I think I didn't get correctly what you were saying [here](https://github.com/sympy/sympy/pull/14161#pullrequestreview-95665411).